numba/np/arrayobj.py

```diff
@@ -3286,10 +3286,19 @@ def array_is_impl(a, b):
 
     return context.compile_internal(builder, array_is_impl, sig, args)
 
+
+@lower_builtin(operator.is_, types.Record, types.Record)
+def record_is(context, builder, sig, args):
+    aty, bty = sig.args
+    if aty != bty:
+        return cgutils.false_bit
+
+    return builder.icmp_unsigned('==', args[0], args[1])
```

Comment:
I don't think this is quite correct. `icmp_unsigned` stands for integer unsigned comparison, and both `args[0]` and `args[1]` are not integers. Your code will fail at a later stage with a similar error:

```
...
numba/core/lowering.py:210: in lower
    self.library.add_ir_module(self.module)
numba/core/codegen.py:727: in add_ir_module
    ll_module = ll.parse_assembly(ir)
../../micromamba/envs/numba/lib/python3.10/site-packages/llvmlite/binding/module.py:25: in parse_assembly
    raise RuntimeError("LLVM IR parsing error\n{0}".format(errmsg))
E   RuntimeError: LLVM IR parsing error
E   <string>:55:20: error: icmp requires integer operands
E     %".40" = icmp eq {i8*, i8*, i64, i64, i32*, [2 x i64], [2 x i64]} %"inserted.strides", %"inserted.strides.1"
E                      ^
```

Think of how you would implement the same operation in pure Python. Perhaps it will help to see how the same is implemented for arrays in this same file.